### DISCUSSION TITLE
How to downgrade version without any data loss?

### DISCUSSION THREAD

**Discussion by TakumiHaruta** [zitadel/zitadel#5517] [Q&A] [answered]
According to https://zitadel.com/docs/self-hosting/manage/updating_scaling#updating-zitadel, `zitadel setup` command executes data migration for the new version, but how do you normally downgrade the version when there is a degradation caused by the new one?

Does `zitadel setup` also handle that or are there any other equivalents to execute data migration for downgrade without any data loss including all events after the Zitadel was upgraded?

[Edited]

Now I noticed there is a new doc for updating, but still not sure the process of downgrading

https://zitadel.com/docs/concepts/architecture/solution#zero-downtime-updates

**Comment by eliobischof** [comment, marked as answer]
Downgrading is as simple as just running the old version, as old data is not modified or deleted when upgrading (yet).
I'll write that with a bit more details to the docs.

**Reply by TakumiHaruta** [reply]
I see but let me make sure about the backward consistency.
To speak of extremes, let's say we run v2.9.0 on an instance but the database has been migrated for v2.22.2, is the app still able to be running?

**Reply by eliobischof** [reply]
By design, yes. Problems could arise with v2.9.0 if incompatible events were added with v2.22.2. Like, a users password can't be decrypted anymore if it was encrypted with an algorithm only known to v2.22.2.

I didn't test that yet, though.

**Reply by TakumiHaruta** [reply]
That's nice! Thanks for your answer
